apps/judicial-system/web/src/utils/formHelper.ts

```diff
@@ -169,3 +171,145 @@ export const hasDateChanged = (
   }
   return false
 }
+
+type stepValidationsType = {
+  [constants.CREATE_RESTRICTION_CASE_ROUTE]: boolean
+  [constants.CREATE_TRAVEL_BAN_ROUTE]: boolean
+  [constants.RESTRICTION_CASE_DEFENDANT_ROUTE]: boolean
+  [constants.RESTRICTION_CASE_HEARING_ARRANGEMENTS_ROUTE]: boolean
+  [constants.RESTRICTION_CASE_POLICE_DEMANDS_ROUTE]: boolean
+  [constants.RESTRICTION_CASE_POLICE_REPORT_ROUTE]: boolean
+  [constants.CREATE_INVESTIGATION_CASE_ROUTE]: boolean
+  [constants.INVESTIGATION_CASE_DEFENDANT_ROUTE]: boolean
+  [constants.INVESTIGATION_CASE_HEARING_ARRANGEMENTS_ROUTE]: boolean
+  [constants.INVESTIGATION_CASE_POLICE_DEMANDS_ROUTE]: boolean
+  [constants.INVESTIGATION_CASE_POLICE_REPORT_ROUTE]: boolean
+  [constants.INVESTIGATION_CASE_CASE_FILES_ROUTE]: boolean
+  [constants.INDICTMENTS_DEFENDANT_ROUTE]: boolean
+  [constants.INDICTMENTS_POLICE_CASE_FILES_ROUTE]: boolean
+  [constants.INDICTMENTS_CASE_FILE_ROUTE]: boolean
+  [constants.INDICTMENTS_PROCESSING_ROUTE]: boolean
+  [constants.INDICTMENTS_CASE_FILES_ROUTE]: boolean
+  [constants.RESTRICTION_CASE_RECEPTION_AND_ASSIGNMENT_ROUTE]: boolean
+  [constants.RESTRICTION_CASE_COURT_OVERVIEW_ROUTE]: boolean
+  [constants.RESTRICTION_CASE_COURT_HEARING_ARRANGEMENTS_ROUTE]: boolean
+  [constants.RESTRICTION_CASE_RULING_ROUTE]: boolean
+  [constants.RESTRICTION_CASE_COURT_RECORD_ROUTE]: boolean
+  [constants.INDICTMENTS_RECEPTION_AND_ASSIGNMENT_ROUTE]: boolean
+  [constants.INVESTIGATION_CASE_RECEPTION_AND_ASSIGNMENT_ROUTE]: boolean
+  [constants.INVESTIGATION_CASE_OVERVIEW_ROUTE]: boolean
+  [constants.INVESTIGATION_CASE_COURT_HEARING_ARRANGEMENTS_ROUTE]: boolean
+  [constants.INVESTIGATION_CASE_RULING_ROUTE]: boolean
+  [constants.INVESTIGATION_CASE_COURT_RECORD_ROUTE]: boolean
+  [constants.INDICTMENTS_OVERVIEW_ROUTE]: boolean
+  [constants.INDICTMENTS_RECEPTION_AND_ASSIGNMENT_ROUTE]: boolean
+  [constants.INDICTMENTS_SUBPOENA_ROUTE]: boolean
+  [constants.INDICTMENTS_PROSECUTOR_AND_DEFENDER_ROUTE]: boolean
+}
+
+export const stepValidations = (theCase: Case): stepValidationsType => {
+  return {
+    [constants.CREATE_RESTRICTION_CASE_ROUTE]: validations.isDefendantStepValidRC(
+      theCase,
+      theCase.policeCaseNumbers,
+    ),
+    [constants.CREATE_TRAVEL_BAN_ROUTE]: validations.isDefendantStepValidRC(
+      theCase,
+      theCase.policeCaseNumbers,
+    ),
```

Comment:
We do not need to run all the validation functions until the `caller` of this function has decided which routes to validate.
```suggestion
export const stepValidations = (): stepValidationsType => {
  return {
    [constants.CREATE_RESTRICTION_CASE_ROUTE]: (theCase: Case) => validations.isDefendantStepValidRC(
      theCase,
      theCase.policeCaseNumbers,
    ),
    [constants.CREATE_TRAVEL_BAN_ROUTE]: (theCase: Case) validations.isDefendantStepValidRC(
      theCase,
      theCase.policeCaseNumbers,
    ),
```